### PR TITLE
feat(multilingualism): 'Requested delivery date' field improvements

### PR DESCRIPTION
**Pull request by netlooker**

### CODE REVIEW

**Comment by cyberwolf** [comment]
Looks good.
Please note I've tested locally wether the hardcoded dates won't cause any issues when they are in the past or outside of the year range (they will be in the future). This did not cause the tests to fail, the date field does not validate this currently, and this was not in scope of this ticket.

**Comment by cyberwolf** [comment]
👍

**Comment by drupol** [comment]
:rainbow:

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module [hunk 1]
```diff
@@ -205,6 +205,15 @@ function _tmgmt_poetry_form_load_translator_settings(&$form_state, &$form, $job)
     $settings = [];
   }
 
+  // NEPT-62: Overwriting the 'delai' field properties.
+  if (isset($settings['delai'])) {
+    $settings['delai']['#title'] = t('Requested delivery date');
+    $settings['delai']['#date_year_range'] = '-0:+1';
```
Comment (tatarbj) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: originally it's -3;+3 https://github.com/ec-europa/platform-dev/blob/a8b9d9625be4d6109a05cb03430edad89e70d37e/profiles/common/modules/custom/tmgmt_poetry/plugins/tmgmt_poetry.ui.inc#L310 why you change it here? the ticket is about making it required only, isn't it?

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module [hunk 2]
```diff
@@ -205,6 +205,15 @@ function _tmgmt_poetry_form_load_translator_settings(&$form_state, &$form, $job)
...
+    $settings['delai']['#default_value'] = '';
```
Comment (tatarbj) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: also the default value is set previously and you overwrite it here to an empty string!

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module [hunk 3]
```diff
@@ -205,6 +205,15 @@ function _tmgmt_poetry_form_load_translator_settings(&$form_state, &$form, $job)
...
+    $settings['delai']['#required'] = TRUE;
+    $settings['delai']['#weight'] = -10;
```
Comment (tatarbj) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: the ticket was not about making it in the top with setting up the #weight (it doesn't matter that much, but still the change makes more than it's expected!)

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module (hunk 1 above)
Comment (Fefaine) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: You cannot request a translation in the past.

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module (hunk 2 above)
Comment (Fefaine) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: Yes, the ticket is to force the user to choose a date in the future, not provide one by default

Comment (tatarbj) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: ok

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module (hunk 3 above)
Comment (tatarbj) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: so it didn't change, you kept it, ok for me.

**Comment by tatarbj** [comment]
👍

File: profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module [hunk 4]
```diff
@@ -205,6 +205,15 @@ function _tmgmt_poetry_form_load_translator_settings(&$form_state, &$form, $job)
     $settings = [];
   }
 
+  // NEPT-62: Overwriting the 'delai' field properties.
+  if (isset($settings['delai'])) {
```
Comment (imanoleguskiza) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: I don't really understand why we overwrite the field properties when we are actually defining the form ourselves. Shouldn't this change be generic for any request?

Comment (Fefaine) [inline profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module]: Actually we want to override the label. The date can be changed by DGT, and the expected delivery date won't be the requested delivery date if that is changed.
So if it was set, we change the label to differenciate this with the date that will show in the jobs

File: profiles/common/modules/custom/tmgmt_poetry/plugins/tmgmt_poetry.ui.inc [hunk 5]
```diff
@@ -312,9 +312,11 @@ class TMGMTPoetryTranslatorUIController extends TMGMTDefaultTranslatorUIControll
     $form['delai'] = array(
       '#type' => 'date_popup',
       '#title' => t('Requested delivery date'),
-      '#default_value' => (!empty($job->settings['delai']) ? $job->settings['delai'] : date('Y-m-d', strtotime('+1 month'))),
+      '#default_value' => '',
       '#date_format' => 'd/m/Y',
-      '#date_year_range' => '-3:+3',
+      '#date_year_range' => '-0:+1',
+      '#required' => TRUE,
+      '#weight' => -10,
```
Comment (voidtek) [inline profiles/common/modules/custom/tmgmt_poetry/plugins/tmgmt_poetry.ui.inc]: Was weight change included in the scope?

File: profiles/common/modules/custom/tmgmt_poetry/src/tmgmt_poetry.behat.inc [hunk 6]
```diff
@@ -228,6 +228,16 @@ class TmgmtPoetrySubContext extends RawDrupalContext implements DrupalSubContext
   }
 
   /**
+   * Generate a relative date for job deadline.
+   *
+   * @When I fill in :arg1 with a relative date of :arg2 days
+   */
+  public function iFillInWithArelativeDateOfDays($field, $number) {
+    $relative_date = date('d/m/Y', strtotime($number));
+    $this->getSession()->getPage()->fillField($field, $relative_date);
```
Comment (voidtek) [inline profiles/common/modules/custom/tmgmt_poetry/src/tmgmt_poetry.behat.inc]: Could insert here a check:

// Retrieve the specified field.
    if (!$field = $this->getSession()->getPage()->findField($field)) {
      throw new ExpectationException("Field '$field' not found.", $this->getSession());
    }

File: profiles/common/modules/custom/tmgmt_poetry/plugins/tmgmt_poetry.ui.inc (hunk 5 above)
Comment (netlooker) [inline profiles/common/modules/custom/tmgmt_poetry/plugins/tmgmt_poetry.ui.inc]: The date field is the only mandatory field that you need to fill. To improve UX after internal discussion we decided to have this field as a first in the field group.

File: profiles/common/modules/custom/tmgmt_poetry/src/tmgmt_poetry.behat.inc (hunk 6 above)
Comment (netlooker) [inline profiles/common/modules/custom/tmgmt_poetry/src/tmgmt_poetry.behat.inc]: Good point. Thanks for the suggestion.

**Comment by dxvargas** [comment]
👍

**Comment by dxvargas** [comment]
👍

**Comment by voidtek** [comment]
:+1:

**Comment by dxvargas** [comment]
👍

**Comment by voidtek** [comment]
:+1:

**Comment by netlooker** [comment]
I've noticed that one test failed but on local there is no problem and the committed code does not touch it. I consider that as a Jenkins environment issue I will investigate that during the merge of the NEPT-182.
cc: @Fefaine

**Pull request merged**